File: app/src/main/java/fr/vpm/mypix/local/FocalLengthLogger.java
package fr.vpm.mypix.local;

import android.content.Context;
import android.util.Log;

import java.util.List;

import fr.vpm.mypix.album.Album;
import fr.vpm.mypix.album.LocalPicture;

public class FocalLengthLogger {

    public void logFocalLength(Context context, List<Album> albums) {
        long startTime = System.currentTimeMillis();
        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.N) {
            albums.forEach(album ->
                    album.getPictures().forEach(picture -> {
                                if (picture instanceof LocalPicture) {
                                    ((LocalPicture) picture).logFocalLength(context);
                                }
                            }
                    ));
        }
        long endTime = System.currentTimeMillis();
        long computingDuration = endTime - startTime;
        Log.d("ux-pic", "Took " + computingDuration + "ms to find EXIF data");
    }
}

File: app/src/main/java/fr/vpm/mypix/local/FocalLengthLogger.kt
package fr.vpm.mypix.local

import android.content.Context
import android.os.Build
import android.util.Log
import fr.vpm.mypix.album.Album
import fr.vpm.mypix.album.LocalPicture
import fr.vpm.mypix.album.Picture

class FocalLengthLogger {
    fun extractExifInfo(context: Context?, albums: List<Album>): List<ExifInfo> {
        var exifInfos = emptyList<ExifInfo>()
        val startTime = System.currentTimeMillis()
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
            exifInfos = albums.flatMap { album: Album ->
                album.pictures.mapNotNull { picture: Picture? ->
                    return@mapNotNull if (picture is LocalPicture) {
                        picture.extractExifInfo(context)
                    } else {
                        null
                    }
                }
            }
        }
        val endTime = System.currentTimeMillis()
        val computingDuration = endTime - startTime
        Log.d("ux-pic", "Took " + computingDuration + "ms to find EXIF data")
        return exifInfos
    }
}